pkg/controller/gather_commands.go

```diff
@@ -494,6 +492,25 @@ func createRemoteConfigConditions(
 	return
 }
 
+// getCustomStoragePath determines a custom storage path by checking configuration sources
+// in priority order:
+// * DataGather CR specification (PersistentVolume.MountPath)
+// * ConfigMap configuration (DataReporting.StoragePath)
+func getCustomStoragePath(configAggregator configobserver.Interface, dataGatherCR *insightsv1alpha2.DataGather) string {
+	if dataGatherCR.Spec.Storage != nil && dataGatherCR.Spec.Storage.Type == insightsv1alpha2.StorageTypePersistentVolume {
+		if storagePath := dataGatherCR.Spec.Storage.PersistentVolume.MountPath; storagePath != "" {
+			return storagePath
+		}
+	}
+
+	// this could be returned directly with the same functionality but this way is more readable
+	if cmsp := configAggregator.Config().DataReporting.StoragePath; cmsp != "" {
+		return cmsp
+	}
+
+	return ""
```

Comment:
I think it is a corner case, but what do you think about returning here the default `/var/lib/insight-operator` path here?